src/plugins/intel_cpu/src/emitters/snippets/x64/jit_loop_emitters.cpp

```diff
@@ -133,118 +114,72 @@ ov::snippets::lowered::ExpressionPtr jit_loop_end_emitter::get_loop_begin_expr(c
-        h->cmp(reg_work_amount, wa_increment);
-        h->jge(*loop_begin_label);
-    }
-
-    for (size_t idx = 0; idx < data_ptr_reg_idxs.size(); idx++) {
-        if (!is_incremented[idx] || finalization_offsets[idx] == 0)
-            continue;
-        Reg64 data_reg = Reg64(static_cast<int>(data_ptr_reg_idxs[idx]));
-        h->add(data_reg, finalization_offsets[idx] * data_sizes[idx]);
-    }
-}
-
-jit_loop_end_dynamic_emitter::jit_loop_end_dynamic_emitter(dnnl::impl::cpu::x64::jit_generator* h, dnnl::impl::cpu::x64::cpu_isa_t isa,
-                                                           const ov::snippets::lowered::ExpressionPtr& expr)
-    : jit_loop_end_emitter(h, isa, expr), loop_end_label{new Xbyak::Label()} {
-    const auto loop_end = ov::as_type_ptr<snippets::op::LoopEndDynamic>(expr->get_node());
-    OV_CPU_JIT_EMITTER_ASSERT(loop_end != nullptr, "expected LoopEndDynamic expr");
-    loop_id = loop_end->get_id();
-
-    const auto begin_expr = get_loop_begin_expr(expr);
-    const auto& loop_begin_emitter = std::dynamic_pointer_cast<jit_loop_begin_dynamic_emitter>(begin_expr->get_emitter());
-    OV_CPU_JIT_EMITTER_ASSERT(loop_begin_emitter, "LoopBeginDynamic expected jit_loop_begin_dynamic_emitter");
-    loop_begin_emitter->set_loop_end_label(loop_end_label);
-}
-
-void jit_loop_end_dynamic_emitter::validate_arguments(const std::vector<size_t> &in, const std::vector<size_t> &out) const {
-    OV_CPU_JIT_EMITTER_ASSERT(loop_end_label != nullptr && loop_begin_label != nullptr, "has not inited labels!");
-    // Note: there must be additional input argument for runtime parameters
-    const auto io_size = num_inputs + num_outputs;
-    OV_CPU_JIT_EMITTER_ASSERT(in.size() == io_size + 1, "Invalid number of in arguments: expected ", io_size + 1, " got ", in.size());
-    OV_CPU_JIT_EMITTER_ASSERT(out.size() == 0, "Invalid number of out arguments: expected ", 0, " got ", out.size());
-}
-
-void jit_loop_end_dynamic_emitter::emit_code(const std::vector<size_t> &in, const std::vector<size_t> &out,
-                                             const std::vector<size_t> &pool_vec_idxs, const std::vector<size_t> &pool_gpr_idxs) const {
-    validate_arguments(in, out);
-    jit_emitter::emit_code(in, out);
-}
-
-void jit_loop_end_dynamic_emitter::emit_impl(const std::vector<size_t>& in, const std::vector<size_t>& out) const {
-    Xbyak::Reg64 reg_runtime_params = abi_param1;  // defined by jit_kernel_emitter
-    Xbyak::Reg64 reg_work_amount = Xbyak::Reg64(static_cast<int>(in[in.size() - 1]));
-    Xbyak::Reg64 reg_increments = Xbyak::Reg64(static_cast<int>(aux_gpr_idxs[0]));
     const auto id_offset = loop_id * sizeof(jit_snippets_call_args::loop_args_t);
+    Reg64 reg_increments = are_ptr_shifts_dynamic ? Reg64(static_cast<int>(aux_gpr_idxs[0])) : Reg64();
+
+#define APPLY_INCREMENTS(USE_RUNTIME_ARGS, TYPE, INCREMENTS, SCALE)                                                         \
```

Comment:
In general I agree)))
But it looks so pretty in the code and so ugly in macro body xD

We have to use macro here because of `offset_of`.
But increments of pointer we can move to lambda function, for sure. Honestly it was so before, but I decided to combine them to one macro. But I agree that it can be difficult for debugging.

I hope now it's better

Thanks!

Reply:
Yes, it's better, thanks 

However, I don't think that we really have to use macro here:
> We have to use macro here because of offset_of

Since the result of `GET_OFF_LOOP_ARGS(m_finalization_offsets)` is just a size_t, we can always pass it through an extra argument:
`apply_increments(bool use_runtime_args, size_t loop_args_offset, const std::vector<int64_t>& increments, size_t scale)`
and then call like:
`apply_increments(are_final_offsets_dynamic, GET_OFF_LOOP_ARGS(m_finalization_offsets), finalization_offsets, 1)`

Not saying we should necessarily do it now, just arguing that it can be done 😁

Reply:
Omg, it's so obvious but I didn't think about that! Thank you very much!
For sure, it will be better if we remove all macros. So I applied your suggestion. Done!